### PR TITLE
feat: add top clicked in predictive layer

### PR DESCRIPTION
**Pull request by mnavarroespinosa**
EX-6350

### CODE REVIEW

**Comment by tiborux** [comment]
Remove the recommendations from the desktop.vue:

```
    <LocationProvider v-if="!$x.query.search" location="no_query">
      <Recommendations />
    </LocationProvider>
```

With the change that you did, you add a horizontal scroll in the main template.

I know that this is not related to this pr, but i think that we can remove it, what do you think?

**Review by ajperezbau — CHANGES_REQUESTED** [review]

File: src/components/results/recommendations.vue [hunk 1]
```diff
@@ -1,25 +1,24 @@
 <template>
   <div
-    v-if="$x.recommendations.length > 0"
+    v-if="!$x.query.searchBox && $x.recommendations.length > 0"
```
Comment (tiborux) [inline src/components/results/recommendations.vue:3]: Adding this check here, we are not showing the top clicked on no results situations.

File: src/components/results/recommendations.vue [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
-        <BaseGrid
-          #default="{ item: result }"
-          :animation="staggeredFadeAndSlide"
-          :items="recommendations"
-        >
-          <Result :result="result" data-test="recommendation-item" />
-        </BaseGrid>
-      </template>
-    </Recommendations>
+    <SlidingPanel
```
[the remark this answers is not in the document]
Comment (mnavarroespinosa) [reply src/components/results/recommendations.vue]: You're right, I've restored the recommendations component that is used for the no results.

File: src/components/results/recommendations.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (mnavarroespinosa) [reply src/components/results/recommendations.vue]: I've restored this component.

File: src/components/results/recommendations.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (mnavarroespinosa) [reply src/components/results/recommendations.vue]: Sure, thanks!

File: src/components/predictive-layer.vue [hunk 3]
```diff
@@ -6,125 +6,136 @@
...
-          <MinusIcon />
-          <span>{{ $t('historyQueries.title') }}</span>
-        </h1>
-
-        <HistoryQueries
+    <BaseKeyboardNavigation
+      class="x-list x-list--gap-06 x-list__item--expand"
+      :class="$x.device === 'mobile' ? 'x-list--vertical' : 'x-list--horizontal'"
+    >
+      <div class="x-list x-list--vertical x-list--gap-06 x-list__item--expand x-flex-1">
```
[the remark this answers is not in the document]
Comment (mnavarroespinosa) [reply src/components/predictive-layer.vue]: Thanks, I think that works great!

**Comment by mnavarroespinosa** [comment]
> Remove the recommendations from the desktop.vue:
> 
> ```
>     <LocationProvider v-if="!$x.query.search" location="no_query">
>       <Recommendations />
>     </LocationProvider>
> ```
> 
> With the change that you did, you add a horizontal scroll in the main template.
> 
> I know that this is not related to this pr, but i think that we can remove it, what do you think?

You're right! I've removed it from `main.vue`.

File: src/components/predictive-layer/custom-recommendations.vue [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<template>
```
Comment (tiborux) [inline src/components/predictive-layer/custom-recommendations.vue:1]: Can we name the component: `sliding-recommendations`?

**Comment by ajperezbau** [comment]
> > Remove the recommendations from the desktop.vue:
> > ```
> >     <LocationProvider v-if="!$x.query.search" location="no_query">
> >       <Recommendations />
> >     </LocationProvider>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > With the change that you did, you add a horizontal scroll in the main template.
> > I know that this is not related to this pr, but i think that we can remove it, what do you think?
> 
> @tiborux You're right! I've removed it from `main.vue`.

Now that the recommendations component is as before it can be added to the main component again without adding the horizontal scroll.

**Comment by ajperezbau** [comment]
The last commit is to fix the sliding panel throwing an error where it doesn't contain an element.

I think the `if` checking the totalResults is not needed, but if it is, it can be placed in the wrapper `div`.

Screenshot 2022-06-23 at 09 39 18

**Review by ajperezbau — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**